src/nvim/ex_getln.c

```diff
@@ -1780,24 +1780,29 @@ getexmodeline (
         msg_col = startcol;
         msg_clr_eos();
         line_ga.ga_len = 0;
-        continue;
+        goto redraw;
       }
 
+      int num_spaces;
       if (c1 == Ctrl_T) {
         int sw = get_sw_value(curbuf);
 
         p = (char_u *)line_ga.ga_data;
         p[line_ga.ga_len] = NUL;
         indent = get_indent_str(p, 8, FALSE);
-        indent += sw - indent % sw;
+        num_spaces = sw - indent % sw;
+        indent += num_spaces;
 add_indent:
-        while (get_indent_str(p, 8, FALSE) < indent) {
+        if (num_spaces > 0) {
+          ga_grow(&line_ga, num_spaces + 1);
```

Comment:
This basically fixes the `CTRL_T` problem. We also need space for the `NUL` that is shifted.